mobile-app/lib/ui/views/learn/challenge_editor/challenge_model.dart

```diff
@@ -93,37 +93,51 @@ class ChallengeModel extends BaseViewModel {
     notifyListeners();
   }
 
+  set setChallenge(Future<Challenge> challenge) {
+    _challenge = challenge;
+    notifyListeners();
+  }
+
+  void init(String url, String? name) async {
+    Challenge challenge = await initChallenge(url, name ?? '');
+
+    setChallenge = initChallenge(url, name ?? '');
```

Comment:
Could you change this so `initChallenge` is called once as it is redundant calling the same function twice right below each other